.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1079,13 +1106,18 @@ private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject refe
                 er[1].Text = weekPrefix + " " + er[1].Text;
             }
 
+            var dateContext = GetYearContext(er[0].Text, er[1].Text, text);
```

Comment:
Is there a way to know if we really need to extract context earlier and skip this call if not?

Reply:
Sorry, I didn't quite get the point.